rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java

```diff
@@ -693,6 +706,157 @@ void getOldFederationBtcUTXOs_calledTwice_returnCachedUtxos() {
 
     }
 
+    @Test
+    void getNewFederation_initialVersion() {
+        List<Integer> storageCalls = new ArrayList<>();
+        List<Integer> deserializeCalls = new ArrayList<>();
+        Federation newFederation = buildMockFederation(100, 200, 300);
+        Repository repositoryMock = mock(Repository.class);
+        FederationStorageProvider federationStorageProvider = createFederationStorageProvider(repositoryMock);
+
+        when(repositoryMock.getStorageBytes(any(
+            RskAddress.class), any(DataWord.class))).then((InvocationOnMock invocation) -> {
+            storageCalls.add(0);
+            RskAddress contractAddress = invocation.getArgument(0);
+            DataWord address = invocation.getArgument(1);
+
+            assertEquals(bridgeAddress, contractAddress);
+
+            if (storageCalls.size() == 1) {
+                // First call is storage version getter
+                Assertions.assertEquals(NEW_FEDERATION_FORMAT_VERSION.getKey(), address);
+                return new byte[0];
+            } else {
+                // Second call is the actual storage getter
+                Assertions.assertEquals(2, storageCalls.size());
+                Assertions.assertEquals(NEW_FEDERATION_KEY.getKey(), address);
+                return new byte[]{(byte) 0xaa};
+            }
+        });
+
+        try (MockedStatic<BridgeSerializationUtils> bridgeSerializationUtilsMocked = mockStatic(BridgeSerializationUtils.class)) {
+            bridgeSerializationUtilsMocked.when(() -> BridgeSerializationUtils.deserializeStandardMultisigFederationOnlyBtcKeys(any(byte[].class), any(NetworkParameters.class))).then((InvocationOnMock invocation) -> {
+                deserializeCalls.add(0);
+                byte[] data = invocation.getArgument(0);
```

Comment:
All this is quite complex, I feel we can simplify a lot by using InMemoryStorage class